### PR TITLE
docs: Document all advanced features in Sphinx guide

### PR DESCRIPTION
**Pull request by naotoo1**
## Summary
- Add comprehensive documentation for 12 previously undocumented features
- Fix incorrect mixed precision example (was showing FCM, now correctly shows GLVQ with `mixed_precision` param)
- Fix incorrect mini-batch example (was showing FCM with `batch_size`, now correctly shows GLVQ)

### Features documented:
- `sample_weight` — per-sample weighted training
- `class_weight` — automatic class balancing
- `validation_data` — validation monitoring during training
- `restore_best` — restore best validation parameters
- `freeze_params` — selective parameter freezing
- `ema_decay` — exponential moving average
- `gradient_accumulation_steps` — gradient accumulation
- `lr_scheduler` — 9 built-in learning rate schedules
- `lookahead` — lookahead optimizer wrapper
- `use_scan` — toggle lax.scan vs Python loop
- `export_predict()` — jax.export for deployment
- `prototype_win_ratios()` — prototype analysis

### CODE REVIEW

**Review by Abubakar26 — APPROVED** [review]
Alles gut
